### ISSUE TITLE
Actions in nonempty patches open as tabs instead of creating/overwriting patches

### ISSUE THREAD

**Issue by solsane** [utk-se/CodeRibbon#85]
If a patch is nonempty, actions that open new panes (ie, create / open file, open settings) create tabs instead of opening as a new patch. 
ie, like what is happening in the lower RHS of this screenshot: https://imgur.com/utqoTUk

**Comment by robobenklein** [comment]
Which solution do you prefer?

 1. Patch is created within the active column (similar to using "Split Down" on the currently active patch)
 2. Patch is created within the nearest-right column that has an empty patch (might go all the way to the end of the ribbon)
 3. Patch is created within the nearest-right column that has less than the configured patches per column setting (by default will convert a column with only one Patch in focus to one with two)
 4. Patch is created in current or next column if a space is available, else creates a new patch column to the right of your current active patch

**Comment by solsane** [comment]
Here's my thought process:
1. When opening a new patch, I want to ensure that it opens somewhere within my focus. Furthermore, if focus is changed, the movement through the ribbon should be a) consistent and b) minimized (in terms of occurrences and distance)
2. I would imagine that as new patches are added, they should be inserted right (and thus the ribbon has an implicit chronological ordering).
3. Adding additional patches shouldn't change how previously added patches are arranged.
4. I should never have to scroll horizontally within a patch

Not a fan of idea 1, for reason 3 (If there's already content and I don't want to split it, I would need to search for a pane that I would want to split) and 2 to a lesser extent.

I think idea 2 might not be great for reason 1. I could see situations where patches are intentionally left blank for 'growing space'. And the inconsistency associated with moving variable number of columns isn't ideal.

Idea  3. Same reasoning as why not a fan of 1 and 2.

TL;DR
4 is my pick, but personally, I would keep it like so: If there is an empty space in the current column, occupy that. Else, insert a new column to the right. 

One additional concern that I'm not exactly sure how I would implement is that not all panes are made equal.
I think for most code patches, the default behavior and tiling is good. But opening settings, markdown preview, or dealing with large blocks of code, among other big things that would occupy a patch, I personally would want that to open with a full column each time. I could see the number of full sized columns be pretty close to the number of columns with 2 or more patches. (Would be a good metric to track!)

When I first imagined this problem, I imagined either (inserting a new column right of my current focus) or inserting a column to the end of the ribbon, and changing focus to it. The latter would offer stronger chronological ordering vs. the first. On the contrary, the user might normally be towards the end of the ribbon anyways, and if they moved to an 'earlier' region in the ribbon, it could indicate that the file is related to that part of the ribbon.
